receiver/azuremonitorreceiver/scraper.go

```diff
@@ -550,6 +588,15 @@ func getMetricAggregations(metricNamespace, metricName string, filters NestedLis
 	return out
 }
 
+func sliceFindInsensitive(slice []string, value string) (int, bool) {
+	for i, v := range slice {
+		if strings.EqualFold(v, value) {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
```

Comment:
You should be able to use one of the `slices.IndexFunc` (or similar) instead of defining a custom function for this.

Reply:
Thank you for the review @MovieStoreGuy I thought about it and doing the EqualFold at scrape time can be avoided creating a copy of the ``Services`` slice config in lower case at start time. I committed an example showing it.

Thinking again I can even go further because of the character exceptional of this hack. I can build booleans on start, to activate or not storageAccount subtypes hack based on services list. I believe it can improve even more readability and performance.

Reply:
I created two commits
- One with the lowercase slice built on start https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/44200/commits/4f1ff2494ec013d17b2e826151a23401004a2b87
- One with simple hack activations booleans built on start https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/44200/commits/aae70a22a7802dafc5c1b1668ab6b6db920eb78c

I let you tell me your pref. Mine is second. Because hack for hack it doesn't need to be dynamically computed.